bazel-java-deps.bzl

```diff
@@ -138,7 +138,7 @@ def install_java_deps():
             "org.sangria-graphql:sangria_{}:2.0.1".format(scala_major_version),
             "org.sangria-graphql:sangria-spray-json_{}:1.0.2".format(scala_major_version),
             "org.scalacheck:scalacheck_{}:1.14.0".format(scala_major_version),
-            "org.scala-lang.modules:scala-collection-compat_{}:2.1.6".format(scala_major_version),
+            "org.scala-lang.modules:scala-collection-compat_{}:2.3.2".format(scala_major_version),
```

Comment:
The old version suffers from an issue in `LazyList` that results in a deadlock in `//ledger/participant-state/kvutils:kvutils-tests_test_suite_src_test_suite_scala_com_daml_
ledger_participant_state_kvutils_export_ProtobufBasedLedgerDataExportSpec.scala`. Afaict, the commit fixing this is https://github.com/scala/scala-collection-compat/pull/350/files but I haven’t looked too closely at the actual fixes upstream.